### DISCUSSION TITLE
Embedded OIDC provider for *The PM Use-Case*

### DISCUSSION THREAD

**Discussion by bobmcwhirter** [guacsec/trustify#31] [Ideas]
Per @ctron in https://github.com/trustification/trustify/issues/11

For "be sure the PM can run it all easily" using an embedded OIDC provider could be beneficial.

A "not for production" feature, necessarily, but "good out-of-the-box experience" feature.

**Comment by ctron** [comment]
Ok, I peeked a bit into [oxide-auth](https://github.com/HeroicKatora/oxide-auth/). I had expected a bit more.

True, one can implement an OIDC provider with it. True, there's is support fox actix-web (and for others). However, in all those cases the support is pretty basic.

My expectation was, that there would be some "out-of-the-box" support for the basic use case. However, it's more a set of building blocks that allow one to create their own OIDC provider, and all of the glue code is missing.

I think it can be done. But I also think it might take a bit of time. And I am not sure it makes sense investing that time at this point, maybe later.

Any thoughts, ideas, alternatives, opinions?

**Reply by ctron** [reply]
In case anyone is interested, I created a draft PR of the current state (no proposing to merge that anytime soon): https://github.com/trustification/trustify/pull/54

Also, the project seems to be responsive. I had to use a patch version due to some crate not being released. That got fixed over night.
